lib/_http_common.js

```diff
@@ -222,9 +222,50 @@ exports.httpSocketSetup = httpSocketSetup;
 /**
  * Verifies that the given val is a valid HTTP token
  * per the rules defined in RFC 7230
+ * See https://tools.ietf.org/html/rfc7230#section-3.2.6
  **/
-const token = /^[a-zA-Z0-9_!#$%&'*+.^`|~-]+$/;
 function checkIsHttpToken(val) {
-  return typeof val === 'string' && token.test(val);
+  if (typeof val !== 'string')
+    return false;
+
+  for (var i = 0, len = val.length; i < len; i++) {
```

Comment:
v8 has been smart enough to cache this automatically for awhile now, so it shouldn't be an issue.